src/main/java/org/primefaces/extensions/optimizerplugin/optimizer/AbstractOptimizer.java

```diff
@@ -93,19 +93,19 @@ protected File aggregateFiles(ResourcesSetAdapter rsa, Charset cset, boolean del
 
                 if (delimeters && outputFile.length() > 0) {
                     // append semicolon to the new file in order to avoid invalid JS code
-                    Files.asCharSink(outputFile, cset, FileWriteMode.APPEND).write(";");
+                    Files.asCharSink(outputFile, cset, FileWriteMode.APPEND).write(System.lineSeparator() + ";");
                 }
 
                 // write / append content into / to the new file
-                Files.asCharSink(outputFile, cset, FileWriteMode.APPEND).write(writer.toString());
+                Files.asCharSink(outputFile, cset, FileWriteMode.APPEND).write(System.lineSeparator() + writer.toString());
```

Comment:
Looks all good to me :+1:  Technically I suppose this adds two line terminators to JS files (?), but there's no problem with that. Regarding the type of line separator, both CSS and EcmaScript support both `\n\r` and `\n`, so that should also be ok.

A micro-optimization (probably not needed): Call `write` twice instead of concatenating `System.lineSeparator()` with `writer.toString()`.

Reply:
good idea just refactored to make it cleaner